### ISSUE TITLE
download-item.cancel() emits done event twice

### ISSUE THREAD

**Issue by dengyaolong** [electron/electron#8625]
* Electron version: 1.4.0
* Operating system: MacOS(Darwin Kernel Version 15.4.0)

### Expected behavior

When I trigger downloadItem.cancel()，event 'done' trigger once that state is 'cancel'

### Actual behavior

event 'done' trigger twice. It's a bug or trigger twice purposely?

Why?

### Reproduce
```
git clone https://github.com/dengyaolong/electron-download-item-cancel-fired-twice
npm i
npm start
```
Save , see the logs. cancel fired  twice
![image](https://cloud.githubusercontent.com/assets/6772690/22722692/e1bd8bba-edf3-11e6-9b49-0b95de61e249.png)

**Comment by deepak1556** [comment]
Thanks for the test case. Its a bug for being called twice, currently when cancel api is called we trigger `content::DownloadItem::Cancel` and `content::DownloadItem::Remove` which will both trigger `content::DownloadItem::OnDownloadItemUpdated` observer. We should the avoid the second call being emitted to the users.

/cc @kevinsawicki

**Comment by nitsakh** [comment]
Hello, I would like to take up this issue. [Not contributed to electron before]

**Comment by Riyuzakii** [comment]
@deepak1556 I am a first timer and i would like to try to contribute. please direct me to the files and tell me what to do.

**Comment by deepak1556** [comment]
@nitsakh @Riyuzakii thanks for your interest! looks like there is already a PR addressing this issue. You can try out other `help/beginner` labeled issues. Thanks!

**Comment by kevinsawicki** [comment]
Fixed by #8678
